src/test/1.6/scala/com/holdenkarau/spark/testing/SampleDatasetTest.scala

```diff
@@ -21,21 +21,6 @@ class SampleDatasetTest extends DatasetSuiteBase {
     equalDatasets(persons, persons)
   }
   
-  test("unequal same schema, different classes") {
```

Comment:
I can't do that in java. I can't get the class tag in java. 
I may add this logic only to Scala and prevent it in java.